File: Python/PE010.py
"""
Problem 10: Summation of primes
URL: https://projecteuler.net/problem=10

    The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

    Find the sum of all the primes below two million.
"""
from shared import primes_generator


######################################################################
# Constants
######################################################################

LIMIT = 2_000_000


######################################################################
# Functions
######################################################################

def sum_primes(limit):
    """
    Sums all the primes bellow a limit number.
    :param limit: The limit number.
    :return: The sum of all the primes.
    """
    result = 0
    primes = primes_generator()
    current = next(primes)
    while current < limit:
        result += current
        current = next(primes)
    return result


def main():
    """
    Main entry for the problem solver.
    """
    result = sum_primes(LIMIT)

    # Show the final result of the problem:
    print(f"The sum of all the primes below {LIMIT} is {result}.")


######################################################################
# Module execution
######################################################################

if __name__ == "__main__":
    main()

File: Erlang/src/pe010.erl
%%%======================================================================
%%% @author Gorka Suárez García
%%% @copyright (C) 2022-2023, Gorka Suárez García
%%% @doc
%%% The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
%%%
%%% Find the sum of all the primes below two million.
%%% @end
%%%======================================================================
-module(pe010).
-author("Gorka Suárez García").
-export([main/0, result/0]).

-define(LIMIT, 2000000).

%%-----------------------------------------------------------------------
%% @doc
%% Main entry for the problem solver.
%% @end
%%-----------------------------------------------------------------------
main() ->
    io:format("The sum of all the primes below ~p is ~p.~n", [?LIMIT, result()]).

%%-----------------------------------------------------------------------
%% @doc
%% Main result for the problem solver.
%% @end
%%-----------------------------------------------------------------------
result() ->
    tools:reduce_while(
        primes, 0, ?LIMIT,
        fun (R, X) ->
            %io:format("~p + ~p~n", [R, X]),
            R + X
        end
    ).
